pkg/controller/statusmanager/status_manager_test.go

```diff
@@ -359,7 +359,7 @@ func TestStatusManagerSetFromDaemonSets(t *testing.T) {
 
 	// Create minimal DaemonSets
 	dsA := &appsv1.DaemonSet{
-		ObjectMeta: metav1.ObjectMeta{Namespace: "one", Name: "alpha", Generation: 1, ClusterName: "", Labels: sl},
+		ObjectMeta: metav1.ObjectMeta{Namespace: "one", Name: "alpha", Generation: 1, Labels: sl},
```

Comment:
The comment in apimachinery 1.24 says "ClusterName was a legacy field that was always cleared by the system and never used." And we're setting it to `""` anyway... so this seems fine, but it makes me wonder why it's even there at all and if this points to a bug/gap in the test cases? It would be good to dig into that a little...